Update dependencies (4.0.7)

ADBond opened the pull request:
<!--
Thank you for your pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->
Previously had a mismatch between stated dependencies + metadata in pypi package. Fixed in [this PR](https://github.com/moj-analytical-services/splink/pull/2551), so we update the dependencies here to match.

<!--
Please add any other relevant info below:
-->
Also added [pip check test](https://conda-forge.org/docs/maintainer/adding_pkgs/#test) now that we can match the metadata with that in pypi.

ADBond committed:
add pip test now that metadata matches

See https://github.com/moj-analytical-services/splink/pull/2551 - now in on 4.0.7

ADBond committed:
add igraph - now explicitly non-optional in splink

ADBond commented:
@conda-forge-admin, please rerender

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found it was in an excellent condition.

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-webservice.

I tried to rerender for you, but it looks like there was nothing to do.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/13673995737](https://github.com/conda-forge/conda-forge-webservices/actions/runs/13673995737). Examine the logs at this URL for more detail.</sub>

ADBond committed:
package name -> python interface

igraph is the core package, python-igraph is the python interface (pypi package just 'igraph')

ADBond commented:
`...run_test.sh: line 7: pip: command not found`

ADBond committed:
require pip

test failing with `pip: command not found`
